cli/src/pcluster/validators/directory_service_validators.py

```diff
@@ -68,6 +70,31 @@ def _validate(self, domain_name):
             )
 
 
+class PasswordSecretArnValidator(Validator):
+    """PasswordSecretArn validator."""
+
+    def _validate(self, password_secret_arn):
+        """Validate that PasswordSecretArn contains an ARN of a readable secret in AWS Secrets Manager."""
+        try:
+            # We only require the secret to exist; we do not validate its content.
+            AWSApi.instance().secretsmanager.describe_secret(password_secret_arn)
+        except AWSClientError as e:
+            if e.error_code == "ResourceNotFoundExceptionSecrets":
+                self._add_failure("The secrets provided in PasswordSecretArn does not exist.", FailureLevel.ERROR)
```

Comment:
In general the validators should provide "semantic" validation and they shouldn't mention parameters names in the error messages. In this way we are decoupling the semantic validation from the schema.

The error message should be: `The Secret Arn {secret_arn} does not exist.`.

We could also rename the `PasswordSecretArnValidator` to `SecretValidator` in this way we could use the same validator to validate different secrets in future and we can extend it to verify multiple things and not just the arn existence in future.

For the same reason we can rename the param from `password_secret_arn` to `secret_arn`.

Reply:
let's talk about this if needed.

Reply:
Will replace the config. name with the actual secret.

But, I'm not sure that generalizing `PasswordSecretArnValidator` into `SecretValidator` is the right choice at this point because:
1. In general I think that generalizations should follow the [Rule of Three Strikes](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)). We can reduce it to 2 strikers, but here we have only one :-)
2. Currently, the name of the validator helps users to identify the offending property. In the long term we should consider to include in the error message the list of offending properties in order to decouple validator's name from the offending properties. Nevertheless, for the time being it is not like that.